src/ray/core_worker/transport/direct_task_transport.cc

```diff
@@ -128,21 +163,27 @@ void CoreWorkerDirectTaskSubmitter::OnWorkerIdle(
   } else {
     auto &client = *client_cache_->GetOrConnect(addr.ToProto());
 
-    while (!queue_entry->second.empty() &&
+    while (!current_queue.empty() &&
            lease_entry.tasks_in_flight_ < max_tasks_in_flight_per_worker_) {
-      auto task_spec = queue_entry->second.front();
+      auto task_spec = current_queue.front();
       lease_entry
           .tasks_in_flight_++;  // Increment the number of tasks in flight to the worker
+
+      // Increment the total number of tasks in flight to any worker associated with the
+      // current scheduling_key
+
+      RAY_CHECK(scheduling_key_entry.active_workers_.size() >= 1);
+      scheduling_key_entry.tot_tasks_in_flight++;
+
       executing_tasks_.emplace(task_spec.TaskId(), addr);
       PushNormalTask(addr, client, scheduling_key, task_spec, assigned_resources);
-      queue_entry->second.pop_front();
+      current_queue.pop_front();
     }
 
     // Delete the queue if it's now empty. Note that the queue cannot already be empty
     // because this is the only place tasks are removed from it.
-    if (queue_entry->second.empty()) {
-      task_queues_.erase(queue_entry);
-      RAY_LOG(DEBUG) << "Task queue empty, canceling lease request";
+    if (current_queue.empty()) {
+      RAY_LOG(INFO) << "Task queue empty, canceling lease request";
```

Comment:
I just added the new method! However, I am not sure about attempting to delete at this place, because there would still be some worker in the active_workers set, so calling the new method would tell us that it's not safe to delete the entry yet

Reply:
Ahh gotcha, makes sense!